dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java

```diff
@@ -149,7 +166,7 @@ private Matches findPatternChange(final String uri) {
             }
         }
 
-        return null;
+        throw new DotRuntimeException("Not pattern match found");
```

Comment:
When this method is call we already are sure (or at least we should) that a Url map pattern exists so, if this method return null then the bug is in the loadPattern method and not here. Throwing an exception prevents us from having to validate a null value any where also validate the null would be hiding the true error.